Source/Core/TerrainEncoding.js

```diff
@@ -0,0 +1,361 @@
+/*global define*/
+define([
+        './AttributeCompression',
+        './Cartesian2',
+        './Cartesian3',
+        './ComponentDatatype',
+        './defined',
+        './Matrix3',
+        './Matrix4',
+        './TerrainCompression'
+    ], function(
+        AttributeCompression,
+        Cartesian2,
+        Cartesian3,
+        ComponentDatatype,
+        defined,
+        Matrix3,
+        Matrix4,
+        TerrainCompression
+    ) {
+    "use strict";
+
+    var SHIFT_LEFT_16 = Math.pow(2.0, 16.0);
+    var SHIFT_LEFT_12 = Math.pow(2.0, 12.0);
+    var SHIFT_LEFT_8 = Math.pow(2.0, 8.0);
+
+    /**
+     * Data used to decompress the terrain mesh.
+     *
+     * @alias TerrainEncoding
+     * @constructor
+     *
+     * @param {Number} minimumX The minimum distance in the x direction.
+     * @param {Number} maximumX The maximum distance in the x direction.
+     * @param {Number} minimumY The minimum distance in the y direction.
+     * @param {Number} maximumY The maximum distance in the y direction.
+     * @param {Number} minimumZ The minimum distance in the z direction.
+     * @param {Number} maximumZ The maximum distance in the z direction.
+     * @param {Number} minimumHeight The minimum height.
+     * @param {Number} maximumHeight The maximum height.
+     * @param {Matrix3} fromENU The east-north-up to fixed frame matrix at the center of the terrain mesh.
+     * @param {Boolean} hasVertexNormals If the mesh has vertex normals.
+     *
+     * @private
+     */
+    var TerrainEncoding = function(minimumX, maximumX, minimumY, maximumY, minimumZ, maximumZ, minimumHeight, maximumHeight, fromENU, hasVertexNormals) {
```

Comment:
I don't know if it fits for sure yet, but did you consider using `AxisAlignedBoundingBox` instead of six separate values for min/max x/y/z?